spec/lib/response-handler.spec.js

```diff
@@ -0,0 +1,128 @@
+    URI = require("urijs");
+    responseHandler = require("../../lib/response-handler")(bluebird, error, logger, parseLinkHeader, URI);
+
+    describe(".isErrorCode()", () => {
+        it("returns false if the status code is within the right range", () => {
+            expect(responseHandler.isErrorCode(300)).toBe(false);
+        });
+        it("returns true if the status code is less than 200", () => {
+            expect(responseHandler.isErrorCode(150)).toBe(true);
+        });
+        it("returns true if the status code is greater than 399", () => {
+            expect(responseHandler.isErrorCode(404)).toBe(true);
+        });
+    });
+    describe(".createErrorForResponse()", () => {
+        var errorResponse;
+
+        it("sets the resposne", () => {
+            errorResponse = {
+                error: "ExampleError",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "Example Error Message"
+                    }
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("Example Error Message"));
+        });
+        it("returns any unkown errors", () => {
+            errorResponse = {
+                body: {
+                    code: "anUnkownError"
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("UNKOWN", "UNKOWN"));
+        });
+        it("returns a timeout error", () => {
+            errorResponse = {
+                code: "exampleTimeoutError"
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("ESOCKETTIMEDOUT"));
+        });
+        it("returns known errors", () => {
+            errorResponse = {
+                error: "duplicate_artifact",
+                response: {
+                    body: {
+                        code: "duplicate_artifact",
+                        message: "A duplicate artifact error message"
+                    }
+                }
+            };
+            expect(responseHandler.createErrorForResponse(errorResponse)).toEqual(new Error("A duplicate artifact error message"));
+        });
+    });
+    describe(".handleErrorResponse()", () => {
+        it("throws errors from responseHandler.createErrorForResponse()", () => {
+            spyOn(responseHandler, "createErrorForResponse").andReturn(new Error("SomeError"));
+            expect(() => {
+                responseHandler.handleErrorResponse();
+            }).toThrow();
+        });
+    });
+    describe(".resolveLink()", () => {
+        it("returns a new URI as a string", () => {
+            var link, resolvedLink, url;
+
+            link = "link/example";
+            url = "url/example";
+            resolvedLink = responseHandler.resolveLink(link, url);
+            expect(resolvedLink).toBe("link/url/example");
```

Comment:
I know we are already changing this, however, this doesn't look right at all. Its supposed to resolve relative. That means I would expect this to be `link/example/url/example`.